pkg/util/util.go

```diff
@@ -0,0 +1,19 @@
+package util
+
+import (
+	"fmt"
+
+	"github.com/docker/engine-api/types/container"
+)
+
+// SafeForLoggingContainerConfig returns a string version of the container.Config object
+// with sensitive information (proxy environment variables containing credentials)
+// redacted.
+func SafeForLoggingContainerConfig(config *container.Config) string {
+	origEnv := config.Env
+	strippedEnv := StripProxyCredentials(origEnv)
+	config.Env = strippedEnv
+	s := fmt.Sprintf("%+v", config)
```

Comment:
I figured there might be cases where it's easier to embed the logic of "what to not log" into the function itself and saves us the trouble of deep copying the object.